experimenter/experimenter/experiments/models.py

```diff
@@ -369,26 +369,26 @@ def targeting(self):
             sticky_expressions.append(f"region in {countries}")
 
         if self.is_sticky and sticky_expressions:
-            sticky_clause = "is_already_enrolled"
-            if is_desktop:
-                if self.is_rollout:
-                    sticky_clause = "experiment.slug in activeRollouts"
-                else:
-                    sticky_clause = "experiment.slug in activeExperiments"
-            sticky_expressions_joined = " && ".join(
-                [f"({expression})" for expression in sticky_expressions]
+            expressions.append(
+                make_sticky_targeting_expression(
+                    is_desktop, self.is_rollout, sticky_expressions
+                )
             )
-            sticky_expression = f"({sticky_clause}) || ({sticky_expressions_joined})"
-            expressions.append(sticky_expression)
         else:
             expressions.extend(sticky_expressions)
 
         if prefs := self._get_targeting_pref_conflicts():
-            expressions.extend(f"!('{pref}'|preferenceIsUserSet)" for pref in prefs)
```

Comment:
Would moving these 2 lines of original code above the `is_sticky` check and putting pref conflict checks as part of `sticky_expressions` do what we want and be a bit simpler?
```py
if countries…
  sticky_expressions.append(country stuff)

if pref_conflicts…
  sticky_expressions.append(prefs stuff)

if is_sticky…
```

Reply:
No, the pref conflict check must always be sticky, even if the targeting is not sticky. Otherwise we will see unenrollments when experiments use an `isEarlyStartup=true` feature. Nimbus will flip the pref and when the targeting gets re-evaluated, the pref conflict targeting will fail due to thet pref being set.